### PR TITLE
feat!: represent small values as single bytes

### PR DESCRIPTION
**Pull request by xunilrj**
This PR is just the first of a series that will update the encoding/decoding of Sway types.

At this PR we are updating codec to Sway 0.48 (in the future for now) and how it treats u8 and bool. Now they are encoded (when possible) with just one byte;

It also fixes a limitation of enums tags being u8 (this was a bug);

The test `test_heap_type_in_enums` had to be commented because we are changing our heap types are encoded, and at the moment, they are not working. :(

Changes to `.github/workflows/ci.yml` can be ignore for now and will be removed after https://github.com/FuelLabs/sway/pull/5264 is merged. As the commented tests.

### Checklist
- [x] I have linked to any relevant issues.
- [x] I have updated the documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary labels.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: packages/fuels-core/src/codec/abi_decoder.rs [hunk 1]
```diff
@@ -384,91 +385,91 @@ mod tests {
...
-        ));
-
-        let s = Token::StringArray(StaticStringToken::new("foo".into(), Some(3)));
-
-        let expected: Vec<Token> = vec![foo, u8_arr, b256, s, ss];
-
-        assert_eq!(decoded, expected);
-        Ok(())
-    }
+    // #[test]
```
Comment (Voxelot) [inline packages/fuels-core/src/codec/abi_decoder.rs:388]: Does this need to be updated or removed?

Comment (digorithm) [reply packages/fuels-core/src/codec/abi_decoder.rs:388]: Why was this commented out?

File: packages/fuels-core/src/types/enum_variants.rs [hunk 2]
```diff
@@ -46,33 +47,34 @@ impl EnumVariants {
...
+    pub fn compute_padding_amount_in_bytes(&self, variant_param_type: &ParamType) -> Result<usize> {
+        let enum_width = self
+            .compute_enum_width_in_bytes()
+            .ok_or(error!(InvalidData, "Error calculating enum width in bytes"))?;
+        let biggest_variant_width = enum_width - ENUM_DISCRIMINANT_BYTE_WIDTH;
+        let variant_width = variant_param_type
+            .compute_encoding_in_bytes()
+            .ok_or(error!(
+                InvalidData,
+                "Error calculating pagging amount in bytes"
```
Comment (Br1ght0ne) [inline packages/fuels-core/src/types/enum_variants.rs:75]:
```suggestion
                "Error calculating padding amount in bytes"
```

File: packages/fuels-core/src/types/param_types.rs [hunk 3]
```diff
@@ -164,65 +164,35 @@ impl ParamType {
...
             // `Bytes` type is byte-packed in the VM, so it's the size of an u8
-            ParamType::Bytes | ParamType::String => Ok(Some(std::mem::size_of::<u8>())),
-            ParamType::RawSlice if !top_level_type => {
-                ParamType::U64.compute_encoding_width().map(Some)
-            }
-            ParamType::StringSlice if !top_level_type => Ok(Some(std::mem::size_of::<u8>())),
-            _ => Ok(None),
+            ParamType::Bytes | ParamType::String => Some(std::mem::size_of::<u8>()),
+            ParamType::RawSlice if !top_level_type => ParamType::U64.compute_encoding_in_bytes(),
+            _ => None,
```
Comment (Br1ght0ne) [inline packages/fuels-core/src/types/param_types.rs:173]: `StringSlice` is not present - was it lost?

**Review by segfault-magnet — APPROVED** [review]

**Review by hal3e — DISMISSED** [review]

**Review by digorithm — DISMISSED** [review]

**Review by Salka1988 — APPROVED** [review]

**Review by digorithm — APPROVED** [review]

**Review by MujkicA — APPROVED** [review]

File: packages/fuels-core/src/codec/abi_decoder.rs (hunk 1 above)
Comment (xunilrj) [reply packages/fuels-core/src/codec/abi_decoder.rs]: Fixed.

**Pull request merged**